src/main/kotlin/uk/gov/justice/digital/hmpps/sendlegalmailtoprisonsapi/onetimecode/OneTimeCodeService.kt

```diff
@@ -19,34 +25,44 @@ class OneTimeCodeService(
   fun createAndSendOneTimeCode(email: String, sessionId: String) {
     OneTimeCode(sessionId, oneTimeCodeGenerator.generateOneTimeCode(), email)
       .also { oneTimeCode -> oneTimeCodeRepository.save(oneTimeCode) }
+      .also { oneTimeCodeAttemptsRepository.save(OneTimeCodeAttempts(sessionId, emptySet())) }
```

Comment:
Creating the attempts cache up front makes for cleaner code when we check it.

Reply:
Nice, yes, I like this approach 👍